### PR TITLE
Disable public methods

### PR DESCRIPTION
**Pull request by cristii006**
This is the PR for the first part of #3019 as discussed on Slack https://robotframework.slack.com/archives/C0K0240NL/p1560209961022000. The second PR related to this issue is in progress.

### CODE REVIEW

**Comment by pekkaklarck** [comment]
Cool! I don't have time to look at the PR now and due to holiday season and various other tasks it may take some time before I have that time.

It might be a good idea to have a separate issue that covers only this part of the #3019. What do you think?

**Comment by cristii006** [comment]
Created issue #3221 for this PR.

File: doc/userguide/src/ExtendingRobotFramework/CreatingTestLibraries.rst [hunk 1]
```diff
@@ -469,6 +469,18 @@ For example, the library below implements only keywords
...
 __ https://docs.python.org/tutorial/modules.html#importing-from-a-package
 
+In order to avoid all public methods of a class that are not decorated with
+`@keyword` decorator becoming keywords, class attribute `ROBOT_AUTO_LIBRARY`
+should be set to False.
+
+ .. sourcecode:: python
+
+     class MyLibrary:
+        ROBOT_AUTO_LIBRARY = False
```
Comment (bkhouri) [inline doc/userguide/src/ExtendingRobotFramework/CreatingTestLibraries.rst]: should this be `ROBOT_AUTO_KEYWORDS` instead of `ROBOT_AUTO_LIBRARY`

Comment (cristii006) [reply doc/userguide/src/ExtendingRobotFramework/CreatingTestLibraries.rst]: Good catch! Thanks!

**Comment by pekkaklarck** [comment]
Thanks for a great PR @cristii006!

Few things that should still be done:

- It would be good to add an explicit test for using this functionality with modules. It should just work, but I'd like to make it sure because the forthcoming `@library` decorator won't work with modules and `ROBOT_AUTO_KEYWORDS` is the only way to enable this functionality.

- The docs also need a note that this is a new feature in RF 3.2. They also need a note that with classes using `@library` is generally a better idea than setting this attribute explicitly, but that cannot really be added until the `@library` PR is merged.

This PR is otherwise in so good shape that I think I'll just merge it and do those changes myself.

**Pull request merged**